examples/systemcalls/systemcalls.c

```diff
@@ -93,7 +201,89 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   
 */
 
-    va_end(args);
+
+    if (outputfile != NULL)
+    {
+        fd = open(outputfile, O_WRONLY|O_CREAT, 0644);
+
+        if (fd < 0)
+        {
+            perror("open");
+            return false;
+        }
+
+
+    }
+    else
+    {
+        return false;
+
+    }
+
+    
+
+    pid = fork();
+    
+    // If fork() returns -1 no child process is created.
+    if (pid == -1) // inside parent process
+    {
+        close(fd);
+        va_end(args);
+        return false;
+    }
+    else if (pid == 0) // Inside child process
+    {
+        // redirect STD_OUT to the file we opened.
+        rc = dup2(fd, STDOUT_FILENO);
```

Comment:
@SundarKrishnakumar think about how you could combine logic with the function above instead of duplicating.